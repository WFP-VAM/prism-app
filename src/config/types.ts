import 'reflect-metadata';
import { rawLayers } from '.';
import type { TableKey } from './utils';

// TODO currently unused. Could be harnessed within admin levels key typing
export type BoundaryKey = 'CODE' | 'CODE1' | 'CODE2';

const optionalMetadataKey = Symbol('optional_property');

// Master Layer type definition. All types/classes looking to exhaust cover of all layer types (nso, wms, etc) should extend upon this type via LayerType['type']
export type LayerType =
  | BoundaryLayerProps
  | WMSLayerProps
  | NSOLayerProps
  | ImpactLayerProps
  | PointDataLayerProps;

export type LayerKey = keyof typeof rawLayers;
/**
 * Check if a string is an explicitly defined layer in layers.json
 * @param layerKey the string to check
 */
export const isLayerKey = (layerKey: string): layerKey is LayerKey =>
  layerKey in rawLayers;

/**
 * Decorator to mark a property on a class type as optional. This allows us to get a list of all required keys at
 * runtime for type checking.
 *
 * Note this has nothing to do with how TypeScript compile-time checking, this is *only* for runtime checks.
 * @param target
 * @param propertyKey
 */
export function optional(target: any, propertyKey: string) {
  const existingKeys: string[] =
    Reflect.getMetadata(optionalMetadataKey, target) || [];
  return Reflect.defineMetadata(
    optionalMetadataKey,
    existingKeys.concat(propertyKey),
    target,
  );
}

export function makeRequired(target: any, propertyKey: string) {
  const existingKeys: string[] =
    Reflect.getMetadata(optionalMetadataKey, target) || [];
  return Reflect.defineMetadata(
    optionalMetadataKey,
    existingKeys.filter(v => v !== propertyKey),
    target,
  );
}

// Generic that verifies that type `T` is a class (basically that it has a constructor)
export type ClassType<T> = { new (...args: any): T };
// create a generic type https://jpwilliams.dev/how-to-unpack-the-return-type-of-a-promise-in-typescript
export type AsyncReturnType<T extends (...args: any) => any> =
  // if T matches this signature and returns a Promise, extract
  // U (the type of the resolved promise) and use that, or...
  T extends (...args: any) => Promise<infer U>
    ? U // if T matches this signature and returns anything else, // extract the return value U and use that, or...
    : T extends (...args: any) => infer U
    ? U // if everything goes to hell, return an `any`
    : any;

/*
 * Get an array of required keys for a class.
 *
 * This has 2 requirements:
 * 1. The class must be instantiable with a no-argument constructor
 * 2. Optional keys should be decorated with `@optional` (or else they will be considered required)
 * @param constructor
 */
export function requiredKeysForClassType(constructor: ClassType<any>) {
  const target = new constructor();
  const allKeys = Object.getOwnPropertyNames(target);
  const optionalKeys = Reflect.getMetadata(optionalMetadataKey, target) || [];
  return allKeys.filter(k => !optionalKeys.includes(k));
}

/**
 * Guard function to check whether a given object is an instance of `classType`.
 * @param classType
 * @param maybeType Object to check.
 * @param logErrors Flag to print out a verbose error message to the console if the object fails
 * @param id
 */
export function checkRequiredKeys<T>(
  classType: ClassType<T>,
  maybeType: Record<string, any>,
  logErrors = false,
  id?: string,
): maybeType is T {
  const requiredKeys = requiredKeysForClassType(classType);
  const missingKey = requiredKeys.find(
    k => !Object.prototype.hasOwnProperty.call(maybeType, k),
  );

  if (logErrors) {
    if (missingKey) {
      console.error(
        `Object %o is invalid: Missing required property '${missingKey}'.`,
        maybeType,
      );
    }

    // Log warnings for keys that aren't a part of this definition
    // eslint-disable-next-line new-cap
    const target = new classType();
    const allKeys = Object.getOwnPropertyNames(target);
    Object.keys(maybeType)
      .filter(key => !allKeys.includes(key))
      .forEach(key =>
        console.warn(`Found unknown key '${key}' on config for ${id}`),
      );
  }
  return !missingKey;
}

export type LegendDefinition = {
  value: string | number;
  color: string;
}[];

export type GroupDefinition = {
  name: string;
  // Main layer of a group of layers. Secondary layers will not trigger notifications.
  main: boolean;
};

export type DownloadDefinition = {
  label: string;
  url: string;
};

export type RawDataConfiguration = {
  scale?: number;
  offset?: number;
  noData?: number;
  // Geotiff pixel resolution, in pixels per degree lat/long
  pixelResolution?: number;
};

export class CommonLayerProps {
  id: LayerKey;

  @optional // only optional for boundary layer
  title?: string;

  type: string;
  opacity: number;

  @optional
  dateInterval?: string;

  @optional // only optional for boundary layer
  legend?: LegendDefinition;

  @optional // only optional for boundary layer
  legendText?: string;

  @optional // only optional for boundary layer
  group?: GroupDefinition;

  @optional
  downloads?: DownloadDefinition[];
}

export class BoundaryLayerProps extends CommonLayerProps {
  type: 'boundary';
  path: string; // path to admin_boundries.json file - web or local.
  adminCode: string;
  adminLevelNames: string[]; // Ordered (Admin1, Admin2, ...)
  adminLevelLocalNames: string[]; // Same as above, local to country
}

export class WMSLayerProps extends CommonLayerProps {
  type: 'wms';
  baseUrl: string;
  serverLayerName: string;

  @makeRequired
  title: string;

  @makeRequired
  legend: LegendDefinition;

  @makeRequired
  legendText: string;

  @optional
  additionalQueryParams?: { [key: string]: string };

  @optional
  wcsConfig?: RawDataConfiguration;
}

export class NSOLayerProps extends CommonLayerProps {
  type: 'nso';
  path: string;

  @makeRequired
  title: string;

  @makeRequired
  legend: LegendDefinition;

  @makeRequired
  legendText: string;

  @makeRequired
  adminCode: string;

  @makeRequired
  adminLevel: number;

  @makeRequired
  dataField: string;
}

export class StatsApi {
  url: string;
  zonesUrl: string;
  groupBy: string;
}

export enum AggregationOperations {
  Mean = 'mean',
  Median = 'median',
}

export type ThresholdDefinition = { below?: number; above?: number };

export class ImpactLayerProps extends CommonLayerProps {
  type: 'impact';

  @makeRequired
  title: string;

  @makeRequired
  legend: LegendDefinition;

  @makeRequired
  legendText: string;

  hazardLayer: LayerKey; // not all layers supported here, just WMS layers
  baselineLayer: LayerKey; // not all layers supported here, just NSO layers. Maybe an advanced way to type this?
  threshold: ThresholdDefinition;

  // defaults to 'median'
  @optional
  operation?: AggregationOperations;

  @optional
  api?: StatsApi;
}

export class PointDataLayerProps extends CommonLayerProps {
  type: 'point_data';
  data: string;

  @makeRequired
  title: string;

  @makeRequired
  legend: LegendDefinition;

  @makeRequired
  legendText: string;

  measure: string;
  @optional
  fallbackData?: string;
  // URL to fetch all possible dates from
  dateUrl: string;
}

export type RequiredKeys<T> = {
  [k in keyof T]: undefined extends T[k] ? never : k;
}[keyof T];

// Get the type of a union based on the value (V) and lookup field (K)
export type DiscriminateUnion<
  T,
  K extends keyof T,
  V extends T[K]
> = T extends Record<K, V> ? T : never;

export type LayersMap = {
  [key in LayerKey]: LayerType;
};

export interface LayersCategoryType {
  title: string;
  layers: LayerType[];
  tables: TableType[];
}

export interface MenuItemType {
  title: string;
  icon: string;
  layersCategories: LayersCategoryType[];
}

export interface MenuItemMobileType {
  title: string;
  icon: string;
  layersCategories: LayersCategoryType[];
  expanded: string;
  selectAccordion: (arg: string) => void;
}

export type AvailableDates = {
  [key in
    | WMSLayerProps['serverLayerName']
    | PointDataLayerProps['id']]: number[];
};

export interface ChartConfig {
  type: string;
  category: string;
  stacked?: boolean;
  exclude?: string[];
  data?: string;
  transpose?: boolean;
  fill?: boolean;
}

export class TableType {
  id: TableKey;
  title: string;
  table: string;
  legendText: string;

  @optional
  chart?: ChartConfig;
}

// used for timeline items in date selector
export type DateRangeType = {
  value: number;
  label: string;
  month: string;
  isFirstDay: boolean;
};
