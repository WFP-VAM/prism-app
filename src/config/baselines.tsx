import nsoDisabled from '../data/nso/NSO_Disabled_Admin1_Total.json';
import nsoHerders from '../data/nso/NSO_Herder_HHs_Admin2.json';
import nsoHerdsize from '../data/nso/NSO_Herd_Size_Admin1_LT_200.json';
import nsoChild from '../data/nso/NSO_Child_U5_Admin2.json';
import nsoLivestock from '../data/nso/NSO_Livestock_Count_ths_Admin2.json';
import nsoElderly from '../data/nso/NSO_Single_Elderly_Admin1_Total.json';
import nsoPoverty from '../data/nso/NSO_Poverty_Headcount_Admin1.json';
import nsoPop from '../data/nso/NSO_Population_Admin2_Total.json';

export interface NsoData {
  TBL_ID: string;
  Period: string;
  CODE?: string;
  SCR_MN: string;
  SCR_ENG: string;
  CODE1?: string;
  CODE2?: string | null;
  SCR_MN1: string;
  SCR_ENG1: string;
  SCR_MN2?: string | null;
  SCR_ENG2?: string | null;
  DTVAL_CO: string;
}

export interface NsoDataset {
  DataList: NsoData[];
}

const nsoDatasets = {
  nsoDisabled,
  nsoHerders,
  nsoHerdsize,
  nsoChild,
  nsoLivestock,
  nsoElderly,
  nsoPoverty,
  nsoPop,
};

type DatasetKeys = keyof typeof nsoDatasets;

export function getNSOData(dataset?: string) {
  if (dataset && dataset in nsoDatasets) {
    return nsoDatasets[dataset as DatasetKeys] as NsoDataset;
  }
  return nsoPoverty;
}
