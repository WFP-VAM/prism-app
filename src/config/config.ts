import appConfig from './mongolia/prism.json';
import rawLayers from './mongolia/layers.json';
import rawTables from './mongolia/tables.json';

export { appConfig, rawLayers, rawTables };
