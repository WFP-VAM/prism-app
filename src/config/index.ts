import { has } from 'lodash';

import {
  indonesiaConfig,
  indonesiaRawLayers,
  indonesiaRawTables,
} from './indonesia';

import {
  mongoliaConfig,
  mongoliaRawLayers,
  mongoliaRawTables,
} from './mongolia';

import {
  mozambiqueConfig,
  mozambiqueRawLayers,
  mozambiqueRawTables,
} from './mozambique';

import { myanmarConfig, myanmarRawLayers, myanmarRawTables } from './myanmar';

import {
  tajikistanConfig,
  tajikistanRawLayers,
  tajikistanRawTables,
} from './tajikistan';

type Country =
  | 'indonesia'
  | 'mongolia'
  | 'mozambique'
  | 'myanmar'
  | 'tajikistan';

const DEFAULT: Country = 'mongolia';

const configMap = {
  indonesia: {
    appConfig: indonesiaConfig,
    rawLayers: indonesiaRawLayers,
    rawTables: indonesiaRawTables,
  },
  mongolia: {
    appConfig: mongoliaConfig,
    rawLayers: mongoliaRawLayers,
    rawTables: mongoliaRawTables,
  },
  mozambique: {
    appConfig: mozambiqueConfig,
    rawLayers: mozambiqueRawLayers,
    rawTables: mozambiqueRawTables,
  },
  myanmar: {
    appConfig: myanmarConfig,
    rawLayers: myanmarRawLayers,
    rawTables: myanmarRawTables,
  },
  tajikistan: {
    appConfig: tajikistanConfig,
    rawLayers: tajikistanRawLayers,
    rawTables: tajikistanRawTables,
  },
} as const;

const { REACT_APP_COUNTRY: COUNTRY } = process.env;
const safeCountry = (COUNTRY && has(configMap, COUNTRY)
  ? COUNTRY
  : DEFAULT) as Country;

const { appConfig, rawLayers, rawTables } = configMap[safeCountry];

export { appConfig, rawLayers, rawTables };
