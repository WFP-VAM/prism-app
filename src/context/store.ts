import {
  combineReducers,
  configureStore,
  getDefaultMiddleware,
} from '@reduxjs/toolkit';

import mapStateReduce from './mapStateSlice';
import serverStateReduce from './serverStateSlice';
import tableStateReduce from './tableStateSlice';
import tooltipStateReduce from './tooltipStateSlice';
import notificationStateReduce, {
  errorToNotificationMiddleware,
} from './notificationStateSlice';
import analysisResultStateSlice from './analysisResultStateSlice';

const reducer = combineReducers({
  mapState: mapStateReduce,
  serverState: serverStateReduce,
  tableState: tableStateReduce,
  tooltipState: tooltipStateReduce,
  analysisResultState: analysisResultStateSlice,
  notificationState: notificationStateReduce,
});

export const store = configureStore({
  reducer,
  // TODO: Instead of snoozing this check, we might want to
  // serialize the state
  middleware: getDefaultMiddleware({
    serializableCheck: false,
    immutableCheck: {
      ignoredPaths: ['mapState.layersData', 'analysisResultState.result'],
    },
  }).concat(errorToNotificationMiddleware),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof reducer>;

export type ThunkApi = {
  dispatch: AppDispatch;
  getState: () => RootState;
  // can add 'extra' and 'rejectValue' here if we ever need them
  // https://redux-toolkit.js.org/usage/usage-with-typescript#createasyncthunk
};

// Used as the third type definition for typing createAsyncThunk
// e.g. createAsyncThunk<ReturnType, HandlerType, CreateAsyncThunkTypes>
export type CreateAsyncThunkTypes = {
  dispatch: AppDispatch;
  state: RootState;
};
