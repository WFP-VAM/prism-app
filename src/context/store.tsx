import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import filtersReduce from './filters/filtersSlice';

export const store = configureStore({
  reducer: {
    filters: filtersReduce,
  },
  // TODO: Instead of snoozing this check, we might want to
  // serialize the state
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
});

export type RootState = ReturnType<typeof store.getState>;
