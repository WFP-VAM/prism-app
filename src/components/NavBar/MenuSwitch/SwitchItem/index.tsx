import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  Box,
  createStyles,
  MenuItem,
  Select,
  Switch,
  Typography,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import { LayerKey, LayerType } from '../../../../config/types';
import {
  getDisplayBoundaryLayers,
  LayerDefinitions,
} from '../../../../config/utils';
import {
  refreshBoundaries,
  safeDispatchAddLayer,
  safeDispatchRemoveLayer,
} from '../../../../utils/map-utils';
import {
  layersSelector,
  mapSelector,
} from '../../../../context/mapStateSlice/selectors';
import { useUrlHistory, getUrlKey } from '../../../../utils/url-utils';
import { removeLayer } from '../../../../context/mapStateSlice';
import { useSafeTranslation } from '../../../../i18n';
import { clearDataset } from '../../../../context/datasetStateSlice';

function SwitchItem({ classes, layer }: SwitchItemProps) {
  const { t } = useSafeTranslation();
  const selectedLayers = useSelector(layersSelector);
  const map = useSelector(mapSelector);
  const dispatch = useDispatch();
  const {
    updateHistory,
    appendLayerToUrl,
    removeLayerFromUrl,
  } = useUrlHistory();

  const { id: layerId, title: layerTitle, group } = layer;

  const selected = selectedLayers.some(({ id: testId }) => {
    return (
      testId === layerId || (group && group.layers.some(l => l.id === testId))
    );
  });

  const selectedActiveLayer = selected
    ? selectedLayers.filter(sl => {
        return (
          (group?.activateAll &&
            group?.layers.find(l => l.id === sl.id && l.main)) ||
          (!group?.activateAll && group?.layers.map(l => l.id).includes(sl.id))
        );
      })
    : [];

  const initialActiveLayer =
    selectedActiveLayer.length > 0 ? selectedActiveLayer[0].id : null;

  const [activeLayer, setActiveLayer] = useState(
    initialActiveLayer || (group?.layers?.find(l => l.main)?.id as string),
  );

  const validatedTitle = t(group?.groupTitle || layerTitle || '');

  const toggleLayerValue = (selectedLayerId: string, checked: boolean) => {
    // clear previous table dataset loaded first
    // to close the dataseries and thus close chart
    dispatch(clearDataset());

    const selectedLayer = group
      ? LayerDefinitions[selectedLayerId as LayerKey]
      : layer;

    const urlLayerKey = getUrlKey(selectedLayer);

    if (checked) {
      const updatedUrl = appendLayerToUrl(
        urlLayerKey,
        selectedLayers,
        selectedLayer,
      );

      updateHistory(urlLayerKey, updatedUrl);

      if (
        !('boundary' in selectedLayer) &&
        selectedLayer.type === 'admin_level_data'
      ) {
        refreshBoundaries(map, dispatch);
      }
    } else {
      removeLayerFromUrl(urlLayerKey, selectedLayer.id);
      dispatch(removeLayer(selectedLayer));

      // For admin boundary layers with boundary property
      // we have to de-activate the unique boundary and re-activate
      // default boundaries
      if ('boundary' in selectedLayer) {
        const boundaryId = selectedLayer.boundary || '';

        if (Object.keys(LayerDefinitions).includes(boundaryId)) {
          const displayBoundaryLayers = getDisplayBoundaryLayers();
          const uniqueBoundaryLayer = LayerDefinitions[boundaryId as LayerKey];

          if (
            !displayBoundaryLayers
              .map(l => l.id)
              .includes(uniqueBoundaryLayer.id)
          ) {
            safeDispatchRemoveLayer(map, uniqueBoundaryLayer, dispatch);
          }

          displayBoundaryLayers.forEach(l => {
            safeDispatchAddLayer(map, l, dispatch);
          });
        }
      }
    }
  };

  const handleSelect = (
    event: React.ChangeEvent<{ value: string | unknown }>,
  ) => {
    const selectedId = event.target.value;
    setActiveLayer(selectedId as string);
    toggleLayerValue(selectedId as string, true);
  };

  const menuTitle = group ? (
    <>
      <Typography className={classes.title}>{validatedTitle}</Typography>
      {!group.activateAll && (
        <Select
          className={classes.select}
          classes={{ root: classes.selectItem }}
          value={activeLayer}
          onChange={e => handleSelect(e)}
        >
          {group.layers.map(menu => (
            <MenuItem key={menu.id} value={menu.id}>
              {t(menu.label)}
            </MenuItem>
          ))}
        </Select>
      )}
    </>
  ) : (
    <Typography className={classes.title}>{validatedTitle}</Typography>
  );

  return (
    <Box key={layerId} display="flex" mb={1}>
      <Switch
        size="small"
        color="default"
        className={classes.switch}
        checked={selected}
        onChange={e => toggleLayerValue(activeLayer, e.target.checked)}
        inputProps={{
          'aria-label': validatedTitle,
        }}
      />
      {menuTitle}
    </Box>
  );
}

const styles = () =>
  createStyles({
    title: {
      lineHeight: 1.8,
    },
    select: {
      '&::before': {
        border: 'none',
      },
    },
    selectItem: {
      whiteSpace: 'normal',
      fontSize: 13,
      fontWeight: 300,
      padding: 0,
      marginLeft: 5,
    },
    switch: {
      marginRight: 2,
    },
  });

export interface SwitchItemProps extends WithStyles<typeof styles> {
  layer: LayerType;
}

export default withStyles(styles)(SwitchItem);
