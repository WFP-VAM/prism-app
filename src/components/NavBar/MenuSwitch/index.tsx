import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  Box,
  Button,
  createStyles,
  Grid,
  Switch,
  Typography,
  withStyles,
  WithStyles,
} from '@material-ui/core';
import {
  LayersCategoryType,
  LayerType,
  TableType,
} from '../../../config/types';
import { addLayer, removeLayer } from '../../../context/mapStateSlice';
import { loadTable } from '../../../context/tableStateSlice';
import { layersSelector } from '../../../context/mapStateSlice/selectors';

function MenuSwitch({ classes, title, layers, tables }: MenuSwitchProps) {
  const selectedLayers = useSelector(layersSelector);
  const dispatch = useDispatch();

  const toggleLayerValue = (layer: LayerType) => (
    event: React.ChangeEvent<HTMLInputElement>,
  ) => {
    const { checked } = event.target;
    if (checked) {
      dispatch(addLayer(layer));
    } else {
      dispatch(removeLayer(layer));
    }
  };

  const showTableClicked = (table: TableType) => {
    dispatch(loadTable(table.id));
  };

  return (
    <Grid item key={title} className={classes.categoryContainer}>
      <Typography variant="body2" className={classes.categoryTitle}>
        {title}
      </Typography>
      <hr />

      {layers.map(layer => {
        const { id: layerId, title: layerTitle, group: LayerGroup } = layer;
        if (LayerGroup && !LayerGroup.main) {
          return null;
        }

        const selected = selectedLayers.some(
          ({ id: testId }) => testId === layerId,
        );

        const validatedTitle = LayerGroup ? LayerGroup.name : layerTitle;

        return (
          <Box key={layerId} display="flex" mb={1}>
            <Switch
              size="small"
              color="default"
              checked={selected}
              onChange={toggleLayerValue(layer)}
              inputProps={{
                'aria-label': validatedTitle,
              }}
            />{' '}
            <Typography variant="body1">{validatedTitle}</Typography>
          </Box>
        );
      })}

      {tables.map(table => (
        <Button
          id={table.title}
          key={table.title}
          onClick={() => showTableClicked(table)}
        >
          <Typography variant="body1">{table.title}</Typography>
        </Button>
      ))}
    </Grid>
  );
}

const styles = () =>
  createStyles({
    categoryContainer: {
      marginBottom: 16,
      '&:last-child': {
        marginBottom: 0,
      },
    },

    categoryTitle: {
      fontWeight: 'bold',
      textAlign: 'left',
    },
  });

export interface MenuSwitchProps
  extends LayersCategoryType,
    WithStyles<typeof styles> {}

export default withStyles(styles)(MenuSwitch);
