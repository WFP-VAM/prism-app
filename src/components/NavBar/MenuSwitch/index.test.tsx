import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import MenuSwitch from '.';
import { store } from '../../../context/store';
import { LayerKey, LayersCategoryType } from '../../../config/types';

const props: LayersCategoryType = {
  title: 'Category 1',
  layers: [
    {
      id: 'ID' as LayerKey,
      title: 'layer',
      type: 'wms',
      baseUrl: 'example.com',
      serverLayerName: 'example',
      legend: [],
      opacity: 0.5,
      legendText: 'legendText',
    },
  ],
  tables: [],
};

test('renders as expected', () => {
  const { container } = render(
    <Provider store={store}>
      <MenuSwitch {...props} />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
