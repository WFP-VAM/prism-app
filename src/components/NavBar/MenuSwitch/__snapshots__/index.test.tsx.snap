// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="MuiGrid-root MenuSwitch-categoryContainer-1 MuiGrid-item"
  >
    <mock-typography
      classname="MenuSwitch-categoryTitle-2"
      variant="body2"
    >
      Category 1
    </mock-typography>
    <hr />
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <mock-switch
        checked="false"
        color="default"
        inputprops="[object Object]"
        size="small"
      />
       
      <mock-typography
        variant="body1"
      >
        layer
      </mock-typography>
    </div>
  </div>
</div>
`;
